### PR TITLE
BaSP-MR-45: Trainers component

### PR DESCRIPTION
**Pull request by Bigzufi**
Create a React component that includes the functionalities of: displaying the list of existing trainers in the database, deleting, creating, or editing them.

### CODE REVIEW

File: package-lock.json
Comment (DelfiLT) [inline package-lock.json:1]: remove package-lock.json from your changes

File: src/Components/Trainers/index.js [hunk 1]
```diff
@@ -1,11 +1,315 @@
...
+  const sendTrainer = async (item) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/`, {
+        method: 'POST',
+        body: JSON.stringify(item),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await response.json();
```
Comment (DelfiLT) [inline src/Components/Trainers/index.js:74]: you can make data as an object to use message, data and error without calling data.data/data.message/data.error

File: src/Components/Trainers/index.js [hunk 2]
```diff
@@ -1,11 +1,315 @@
...
+      if (!data.error) {
+        addTrainer(data.data);
+        setToggleAdd(false);
+      } else {
+        throw data.message;
+      }
+    } catch (error) {
+      setErrorMessage(error);
```
Comment (DelfiLT) [inline src/Components/Trainers/index.js:82]: you can use console.error

**Review by SantiH97 — CHANGES_REQUESTED** [review]
Overall the code looks clean and easy to read , Good job! make those small changes and you should be good to go.

File: src/Components/Trainers/index.js (hunk 2 above)
Comment (Bigzufi) [reply src/Components/Trainers/index.js:82]: setErrorMessage is a state used to display validation errors below the form.

File: src/Components/Trainers/index.js [hunk 3]
```diff
@@ -1,11 +1,315 @@
...
+    dni: '',
+    phone: '',
+    email: '',
+    city: '',
+    password: '',
+    salary: ''
+  });
+
+  const getTrainers = async () => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers`);
```
[the remark this answers is not in the document]
Comment (Bigzufi) [reply src/Components/Trainers/index.js:22]: 👍

**Review by SantiH97 — APPROVED** [review]

File: .gitignore
Comment (DelfiLT) [inline .gitignore:1]: You shouldn't change the gitignore file, try to remove the package lock json as Ferpa said

**Review by DelfiLT — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+package-lock.json
```
Comment (Juanoauil) [inline .gitignore:2]: remove this changes

File: src/Components/Trainers/index.js (hunk 2 above)
Comment (Juanoauil) [reply src/Components/Trainers/index.js:67]: you shouldn't do validations, please use console.error

File: src/Components/Trainers/index.js [hunk 5]
```diff
@@ -1,11 +1,321 @@
...
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+      {toggleEdit && (
+        <form onSubmit={editTrainer}>
+          <div>
+            <fieldset>
+              <label htmlFor="">Name</label>
```
Comment (Juanoauil) [inline src/Components/Trainers/index.js:257]: why do you use htmlFor?

File: src/Components/Trainers/index.js (hunk 2 above)
Comment (Bigzufi) [reply src/Components/Trainers/index.js:67]: 👍

File: src/Components/Trainers/index.js (hunk 5 above)
Comment (Bigzufi) [reply src/Components/Trainers/index.js:257]: sorry it's emmet... i will remove it

File: .gitignore (hunk 4 above)
Comment (Bigzufi) [reply .gitignore:2]: 👍

**Review by DelfiLT — APPROVED** [review]

**Review by Dannylez — CHANGES_REQUESTED** [review]
remove package-lock.json from your changes

**Review by Juanoauil — APPROVED** [review]

**Review by Dannylez — APPROVED** [review]

**Pull request merged**